packages/storage/src/providers/AWSS3Provider.ts

```
@@ -446,7 +448,23 @@ export class AWSS3Provider implements StorageProvider {
 				throw error;
 			}
 		}
-
+		if (validateObjectExistence) {
+			try {
+				await s3.send(headObjectCommand);
+			} catch (error) {
```

I'm wondering, do we need to be more precise with the error handling here? For example explicitly checking that the request failed with a 404 before dispatching the event that says `${key} not found`. We'd still want to throw the error regardless of failure type, but just to make sure our eventing is accurate.

---

Sure. Updated.